x-pack/test/api_integration/apis/ml/data_frame_analytics/delete.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/machine_learning/security_common';
+import { DataFrameAnalyticsConfig } from '../../../../../plugins/ml/public/application/data_frame_analytics/common';
+import { DeepPartial } from '../../../../../plugins/ml/common/types/common';
+const COMMON_HEADERS = {
+  'kbn-xsrf': 'some-xsrf-token',
+};
+
+export default ({ getService }: FtrProviderContext) => {
+  const es = getService('legacyEs');
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const jobId = `bm_${Date.now()}`;
+  const generateDestinationIndex = (analyticsId: string) => `user-${analyticsId}_1`;
+  const commonJobConfig = {
+    source: {
+      index: ['ft_bank_marketing'],
+      query: {
+        match_all: {},
+      },
+    },
+    analysis: {
+      classification: {
+        dependent_variable: 'y',
+        training_percent: 20,
+      },
+    },
+    analyzed_fields: {
+      includes: [],
+      excludes: [],
+    },
+    model_memory_limit: '350mb',
+  };
+
+  const testJobConfigs: Array<DeepPartial<DataFrameAnalyticsConfig>> = [
+    'Test delete job only',
+    'Test delete job and target index',
+    'Test delete job and index pattern',
+    'Test delete job, target index, and index pattern',
+  ].map((description, idx) => {
+    const analyticsId = `${jobId}_${idx + 1}`;
+    return {
+      id: analyticsId,
+      description,
+      dest: {
+        index: generateDestinationIndex(analyticsId),
+        results_field: 'ml',
+      },
+      ...commonJobConfig,
+    };
+  });
+
+  async function createJobs(mockJobConfigs: Array<DeepPartial<DataFrameAnalyticsConfig>>) {
+    for (const jobConfig of mockJobConfigs) {
+      await ml.api.createDataFrameAnalyticsJob(jobConfig as DataFrameAnalyticsConfig);
+    }
+  }
+
+  async function jobDeleted(analyticsId: string) {
```

Comment:
I'd rather have this as a service method in `ml.api` with some retry to make is more stable, similar to [`waitForAnomalyDetectionJobNotToExist`](https://github.com/elastic/kibana/blob/master/x-pack/test/functional/services/ml/api.ts#L293)